### PR TITLE
Add time and date pickers

### PR DESCRIPTION
**Pull request by AndreVero**

### CODE REVIEW

File: app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsScreen.kt [hunk 1]
```diff
@@ -46,6 +47,11 @@ fun EventDetailsScreen(
     val context = LocalContext.current
     val snackbarHostState = LocalSnackbarHostState.current
 
+    val dateFromDialog = rememberMaterialDialogState()
+    val timeFromDialog = rememberMaterialDialogState()
+    val dateToDialog = rememberMaterialDialogState()
+    val timeToDialog = rememberMaterialDialogState()
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsScreen.kt:53]: You could just have 2 of these

1- dateDialog
1- timeDialog

And just reuse those
No need to create 2 timeDialog and 2 dateDialog

File: app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.compose.runtime.Composable
+import com.vanpra.composematerialdialogs.MaterialDialog
+import com.vanpra.composematerialdialogs.MaterialDialogState
+import com.vanpra.composematerialdialogs.datetime.time.timepicker
+import java.time.LocalTime
+
+@Composable
+fun TimeDialog(
+    dialogState : MaterialDialogState,
+    onTimePicked : (time: LocalTime) -> Unit
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt:12]: Remember to add a modifier

File: app/src/main/java/com/vero/tasky/agenda/presentation/util/LocalDateParser.kt [hunk 3]
```diff
@@ -31,4 +32,12 @@ object LocalDateParser {
         val parsedMonth = date.month.getDisplayName(TextStyle.SHORT, Locale.getDefault())
         return String.format("$parsedMonth %02d ${date.year}", date.dayOfMonth)
     }
+
+    fun updateLocalDateTime(localDateTime: LocalDateTime, localTime: LocalTime) : LocalDateTime {
+        return localDateTime.withHour(localTime.hour).withMinute(localTime.minute)
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/util/LocalDateParser.kt:37]: Pretty sure you could

localDateTime.of(localDateTime.date, localTime)

File: app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt (hunk 2 above)
Comment (AndreVero) [reply app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt:12]: MaterialDialog (from the library) doesn't have a "modifier" parameter, so I can't use it anywhere

File: app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsScreen.kt (hunk 1 above)
Comment (AndreVero) [reply app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsScreen.kt:53]: Yeah, I thought about that but then stopped on this option, as for me in this way it's more readable code. Or will it degrade performance too much?

Comment (MKiperszmid) [reply app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsScreen.kt:53]: It wont degrade performance
Its just unnecessary haha

File: app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt (hunk 2 above)
Comment (MKiperszmid) [reply app/src/main/java/com/vero/tasky/agenda/presentation/components/TimeDialog.kt:12]: Ah interesting

No problem then

**Pull request merged**